### ISSUE TITLE
FMA

### ISSUE THREAD

**Issue by cmungall** [OBOFoundry/OBOFoundry.github.io#21]
Currently hardcoded to be the ancient obo format version I made. Should be made the complete ontology now it is in OWL. obo-basic can be a subproduct

**Comment by cmungall** [comment]
cc @trapperkeeper

Now FMA natively produces OWL (not valid OWL2 but will be soon) do we just point directly to it? Do we create a distinct OBO-compatible version? (e.g. OBO PURLs, standard annotation properties and object properties)?

**Comment by trapperkeeper** [comment]
Of course my vote, speaking for the FMA team, would be to use the FMA directly. It is confusing having multiple differing versions in play.

**Comment by cmungall** [comment]
What should the status of the OBO purls be? Both class purls and ontology purls?

 1. Keep these active, and have them direct to an 'OBO translation' of FMA
 2. Obsolete them
 3. FMA obsoletes its existing purls and joins the OBO system

We are effectively doing 1 right now. With the new FMA being natively OWL we can update the translation to be a minimal one of rewriting IRIs, thus the OBO shadow will be semantically equivalent

2 may well break lots of things

3 is up to @trapperkeeper but I suspect the answer is no

See also https://github.com/OBOFoundry/purl.obolibrary.org/issues/63#issuecomment-161012932

**Comment by trapperkeeper** [comment]
Sorry for the late response. I brought this issue up with the FMA team but, as suspected, we are not prepared to obsolete our existing purls. First, some of our purls are already in use, as data annotations, referenced from other sources like UMLS, etc. Second, there are multiple community ontology efforts out there, and sometimes professional tension between them. The FMA, in an attempt to remain neutral, has adopted a purl that is not aligned with any specific effort (just trying to remain Switzerland here).  

Of course, there are factors that may affect FMA purls. The future of purl.org being one that we are watching. 

Regarding choices 1 or 2, from our perspective if would, of course, be nice if there were only one set of purls. But, for legacy reasons and to maintain exiting references I can certainly understand that some mapping or translation method may be necessary.

**Comment by cmungall** [comment]
Thanks

Just to point out one thing: purl.obolibrary.org is now completely 
independent of purl.org. You should not use purl.org for anything as the 
future is, as you hinted, in doubt. purl.org could vanish tomorrow and 
the obolibrary ontologies would not be affected

**Comment by trapperkeeper** [comment]
Any purl scheme is a tie to a purl provider, be it purl.org or purl.obolibrary.org. We chose purl.org some time ago because we thought, at the time, that it would be the most universal and neutral provider. We expected the purl user community to embrace the unbranded purl and therefore expected it to be a stable resource. They are now on the ropes, as you say. However, we are already using them. Those FMA purls are already out there in the world. So, until we know its fate for sure, we are holding out to see what happens to purl.org. There is some talk of it being picked up and maintained by w3id.org: 
https://www.jiscmail.ac.uk/cgi-bin/webadmin?A2=ind1511&L=DC-ARCHITECTURE&F=&S=&P=3711 
https://groups.google.com/forum/#!topic/persistenturls/Zpd4BHQxxIM

**Comment by cmungall** [comment]
any more thoughts on this? http://purl.org/sig/ont/fma.owl still doesn't resolve

**Comment by trapperkeeper** [comment]
I can make it resolve to a web accessible copy of the OWL file itself, but it does not specify a version of the ontology. 

 

OK, I have modified the PURL to resolve to the latest version (i.e. http://sig.biostr.washington.edu/share/downloads/fma/release/latest/fma.owl ). 

 

I'm not sure of the context in which this arose, does this fit the bill?

 

-Todd

**Comment by cstoeckert** [comment]
Tested link and its resolving

**Comment by cmungall** [comment]
Not sure why this was closed, the issue is still not resolved. OBO is serving an old version of FMA

**Comment by nlharris** [comment]
who could fix this?

**Comment by cmungall** [comment]
take this to an ops call

**Comment by cmungall** [comment]
this is actually a seriously bad problem

**Comment by dosumis** [comment]
I'd really like this to be fixed. Need the latest FMA for a couple of projects currently.  Can we have this resolve to the obolibrary PURL in addition to any PURL preferred by FMA?

I notice the owl files are now provided zipped.  I think this can work OK with at least some infrastructure if they are names FMA.owl.zip, but they seem to be just FMA.zip.

**Comment by trapperkeeper** [comment]
Hey all, I will stick a static copy of the FMA.owl at that address, as an interim solution.  I can't do it until tonight, but will do so.

**Comment by trapperkeeper** [comment]
I grabbed a few moments away from what I was working on and posted that file. The PURL should work now (at least until the next FMA release which isn't pending).

**Comment by nlharris** [comment]
Thanks, @trapperkeeper!

Can we now close this issue?

**Comment by cmungall** [comment]
unfortunately not

**Comment by matentzn** [comment]
Works now?

**Comment by matentzn** [comment]
```
matentzn@mbp:~/tmp_data $ curl -L -s http://purl.org/sig/ont/fma.owl | head
<?xml version="1.0"?>
<rdf:RDF xmlns="http://purl.org/sig/ont/fma.owl#"
     xml:base="http://purl.org/sig/ont/fma.owl"
     xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
     xmlns:owl="http://www.w3.org/2002/07/owl#"
     xmlns:xsd="http://www.w3.org/2001/XMLSchema#"
     xmlns:rdfs="http://www.w3.org/2000/01/rdf-schema#"
     xmlns:fma="http://purl.org/sig/ont/fma/"
     xmlns:dc="http://purl.org/dc/elements/1.1/">
    <owl:Ontology rdf:about="http://purl.org/sig/ont/fma.owl">
```

```
curl -L http://sig.biostr.washington.edu/share/downloads/fma/release/latest/fma.owl | head
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0<?xml version="1.0"?>
<rdf:RDF xmlns="http://purl.org/sig/ont/fma.owl#"
     xml:base="http://purl.org/sig/ont/fma.owl"
     xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
     xmlns:owl="http://www.w3.org/2002/07/owl#"
     xmlns:xsd="http://www.w3.org/2001/XMLSchema#"
     xmlns:rdfs="http://www.w3.org/2000/01/rdf-schema#"
     xmlns:fma="http://purl.org/sig/ont/fma/"
     xmlns:dc="http://purl.org/dc/elements/1.1/">
    <owl:Ontology rdf:about="http://purl.org/sig/ont/fma.owl">
```

These work now?

**Comment by cmungall** [comment]
OK, PURLs work for me now

the question now is what to do with the class PURLs. We have OBO  class PURLs for FMA from my translation that predates official FMA PURLs. The official FMA purls are not OBO formatted

Some options

 1. Deprecate OBO class PURLs, and "re-admit" FMA into OBO registry with the new PURLs as canonical
 2. Have shadow OBO class PURLs, as we do for NCIT

I am not sure how best to implement 1. I guess a one time release of the old OBO FMA version with all classes obsoleted and replaced by the sig PURLs. This would create an odd situation. the deprecated OBO class PURLs would be maintained within OBO infrastructure but the official FMA purls would exist without.

It's not really clear how meaningful it is to have FMA "in" OBI with 1.

**Comment by nlharris** [comment]
@matentzn does this still need attention from CJM?

**Comment by cmungall** [comment]
I am not sure I am needed for any of this

I already made #1458 but no one commented on it :-).  

I think we just need shared agreement between FMA and OBO

 - does FMA exist outside OBO or within?
 - if in, does FMA retain its current URIs? I assume yes. If so, then OBO will need to adjust its policy perhaps as a grandfather clause only. I think this will have to be discussed on an OBO call
 - alternatively something the NCIT solution could be pursued but I suspect that will be deemed less than satisfactory

**Comment by dosumis** [comment]
Do we need to arrange a call with someone at FMA to finalise and implement agreement on this?  We need resolution for pending work with HCA and HubMap.

**Comment by cmungall** [comment]
I think so but I would recommend HCA/HM just use the native FMA download
PURL and class PURLs, and doesn't depend on OBO registry issues (we'd also
need to coordinate with Uberon, and likely switch the Uberon bridge PURLs)

**Comment by nlharris** [comment]
What is the status of this? Does someone need to relay info to FMA? What about the linked draft PR, DO NOT MERGE: one solution to FMA problem #21 #1458?

**Comment by nlharris** [comment]
@matentzn  does this still need the nuclear "attn:CJM" label?

**Comment by matentzn** [comment]
Nope, I removed it. This is now mostly a @dosumis issue.

**Comment by cmungall** [comment]
Let's just update the entry in obo metadata, unobsolete it and include the following as the text:

_This entry is for the official Foundational Model of Anatomy. Note that this ontology differs from other ontologies in OBO, it does not use OBO PURLs. It may also not have the same QC procedures applied, and the ontology may not be compatible with OWL reasoners. See https://github.com/OBOFoundry/OBOFoundry.github.io/issues/21 for details._

_Note that previously there was a historic OBO conversion of the FMA that used OBO purls from 2009. This can still be used:_

_http://ontologies.berkeleybop.org/fma.owl_

_But note this is never updated, has out of date content, and is not sanctioned by the FMA and may be removed later._



Then at the same time we update purl.obolibrary.org/conf and have fma.owl redirect to

http://sig.biostr.washington.edu/share/downloads/fma/release/latest/fma.owl

We would remove the obo format download.

What we put for the browser is largely irrelevant as there are no more class PURLs to redirect

Is this ideal? Far from it.

 1. we are essentially deleting OBO class PURLs, and adding an ontology to OBO that does not use OBO PURLs
 2. the standards for which annotation properties are used are very different from the rest of OBO
 3. there are vastly different modeling patterns. See for example the branch 'General anatomical term' or 'Deprecated term'
 4. the current OWL version of FMA above is logically incoherent, which will cause problems for any pipeline using a reasoner

However, it is far better than the current situation, and we can implement it immediately, so let's do it, but let's email the FMA group first for confirmation.

Note the Uberon group can work on making new coherent version of FMA that follows OBO practice, for use in projects like hubmap and hca. However, this becomes an uberon / anatomy ontology group problem and we can close this issue.

cc-ing @graybeal this brings us closer to bioportal which is showing the latest fma (5.0.0 from 2019), although bioportal shows the UMLS version which lacks some axioms in the full OWL (which actually has the advantage of being coherent)

As an aside, for information purposes, no action required for OBO purposes, here is an example explanation of an incoherency:

![image](https://user-images.githubusercontent.com/50745/146097338-2187b8ed-dbfd-4277-9db9-86fe003424ff.png)

![image](https://user-images.githubusercontent.com/50745/146097384-324a8bec-1b37-4452-ac23-97eec28821d9.png)

**Comment by matentzn** [comment]
I will leave this up to @dosumis and his team to deal with once they get deeper into Human Anatomy land.

**Comment by lschriml** [comment]
Looping in Melissa Clarkson who is working on the FMA to this conversation.

**Comment by nlharris** [comment]
Are there still actions required to close this ticket?

**Comment by lschriml** [comment]
Has the FMA group responded/provided feedback ?

**Comment by cmungall** [comment]
We need to do something about this situation, we are confusing the community here.

**Comment by cmungall** [comment]
@trapperkeeper - looks like FMA is no longer available from http://sig.biostr.washington.edu/share/downloads/fma/release/latest/fma.owl?

**Comment by cmungall** [comment]
OK, here is a reminder of where we are, and some paths forward.

Currently the version of FMA from OBO class and ontology PURLs is a very old subset I made over a decade ago. We need to deprecate this product ASAP. I made the subset when there was no official FMA OWL, since then the FMA team have made an OWL product available. However:

- it has been difficult getting a stable URL for the FMA, the most recent one http://sig.biostr.washington.edu/share/downloads/fma/release/latest/fma.owl has stopped working. The most reliable PURL is currently the NCBO one
- the current official FMA OWL is incoherent (in the technical sense, this is not pejorative). For many applications of the FMA this is fine, but it obviously causes problems when combined with reasoning
- the official FMA OWL does not use OBO PURLs

# Option 1: Keep a live entry in OBO with a link to bioportal

We would keep the current FMA entry in the metadata, the `products` field would be an empty list, and the textual description would have a link to bioportal and the bioportal download URL. There would be some minimal text explaining the history, explaining the special status of FMA in OBO

Ontology browsers such as OLS and ontobee would not get the official version from an OBO PURL, they would add an additional entry to their own registries.

We would add redirects for the class PURLs to go to a designated ontology browser.

# Option 2: mark entry as deprecated

Mark the entry as deprecated and state that FMA is not part of the OBO library - making it clear this is not a reflection of quality or content, it is purely a technical decision regarding use of a common technical framework.

As for 1, we would add redirects for the class PURLs to go to a designated ontology browser (perhaps with different http redirect code?)

# Option 3: engage the FMA team to use OBO PURLs and make FMA fully part of OBO

Seems unlikely, see this thread

**Comment by matentzn** [comment]
@deepakunni3 can you drive this issue forward?

The first thing is to read the meeting minute notes and figure out what the best way forward is. I think @lschriml wanted to double check what the concrete status today is of FMA (not in the sense of "we have plans to revive FMA at some point in the future, but concrete evidence that it is currently active). If that is negative, we can go with option 2. 

I think no matter what we do, we should do option 1 anyways; I think @deepakunni3 you can make a PR to that end. This is safe, it merely gets rid of the ancient obo FMA. Note there needs to be an update in the purl config as well.

**Comment by lschriml** [comment]
My suggestion was that the OBO Foundry contact and speak with the FMA.
@cmungall -- as you have been leading this task so far, would contact the FMA.

**Comment by brinkleyjf** [comment]
After reading more about how OBO works I have  unzipped fma.zip so that fma.owl is now available directly from http://sig.biostr.washington.edu/share/downloads/fma/release/latest/fma.owl. (The same file is still downloadable from Bioportal). However, the file is 254M so it takes  awhile to download. The zip file, which also contains fma puns, is only 9.6 M. And FMA URIs like http://purl.org/sig/ont/fma/fma62955 (Anatomical Entity) still resolve to our local FMA browser. 

PS For those who don't know me I'm the longtime director of the Structural Informatics Group (SIG) and am now the only one maintaining the SIG  website (http://si.washington.edu), which also has the downloadable FMA and the FMA browser via redirects. If Onard Mejino supplies me with any new versions I'll be the one who adds them to the site. I really like the idea behind the OBO Foundry, especially the idea of linking ontologies and data, so I'll be happy to work with you to help that happen as much as possible, even if we don't have the resources to make the FMA fully compliant with OBO Foundry principles.

**Comment by cmungall** [comment]
Thanks Jim. The 254M is fine, it's better to have the ontology PURL be unzipped as the whole OWL infrastructure works better that way.

Note also that in fixing the sig URL you have also fixed the canonical ontology PURL http://purl.org/sig/ont/fma.owl - thank you!

From the email thread, it's still not entirely clear to me if you are still considering moving to OBO class PURLs, or if that is off the table? It sounds like you do not have resources for this.

I'm also still not entirely clear on the difference between the different versions. Clearly the files from your site are the same regardless of whether compressed or not. But my reading of the thread is that the sig version uses OWL punning whereas the bioportal version does not? It's important for us in OBO to have clarity on the different versions. We can absolutely have different download links in OBO, so long as each product is clearly described. But maybe I misread and all versions are the same?

Note things like punning are not necessarily an issue, but as I mentioned previously, the OWL is unfortunately incoherent (in the technical not pejorative sense). If you run a reasoner you will see many unsatisfiable classes. We can help you resolve these - but we should be sure we have a strategy in place to avoid them popping up again if Onard does make any new releases.

Thanks for your information. Here are the options moving forward

1. We retain the FMA entry in OBO, but extend the OBO metadata so we can have non-OBO PURLs in the system. We have the existing OBO class purls redirect to FMA class PURLs like  http://purl.org/sig/ont/fma/fma62955, and we redirect the OBO ontology PURL to the official purl.org/sig one. We put a note on the page that the ontology is known to have unsatisfiable classes and include instructions on how to repair using ROBOT
2. the OBO/anatomy team creates a variant of the official FMA that has OBO PURLs, fixes the unsats, and potentially rewires some of the relations to RO. We commit to syncing this with any future releases of FMA (which may be unlikely)
3. As 2, but Jim/the SIG team does this [seems unlikely given current resources]

**Comment by brinkleyjf** [comment]
Chris,  some responses:
OBO vs FMA purls
I basically need to understand the pros and cons of this better, and also talk with others like Todd, Onard and Melissa Clarkson. I know the biggest issue in the past was our belief that alot of people are already using the FMA purls, so it would be a pain for them to change. I  don't know how many are actually using them though.   Seems like it would be relatively easy  for you or someone to write a script  to replace the purls in fma.owl with OBO purls, and the FMA would still get credit  since OBO purls have the source ontology in them. But if alot of people are already using fma purls that could be a problem. How important it is for OBO to use only obo purls for the goals of linked data to be achieved? I saw somewhere you have made an exception for NCIT. Could the FMA be another such exception as you suggest in your option number 1?

FMA puns
I am pretty sure fma.owl on our site is the same as in Bioportal, ie no puns. pun_fma.owl is a separate file that can be imported into fma.owl.  I modified the description of fma resources at http://si.washington.edu/content/summary-fma-resources in order to better explain this.

Inconsistencies in reasoning
Seems like these should be removed   even in our current release, but I don't know how hard it would be to do that, or how doing so would affect the content. I'll check with Todd and Onard to see what they did. My vague recollection is that the reasoner never completed, but maybe they fixed that. Maybe your ROBOT tools could do better.

Your options
These seem to be about the same as your earlier ones, except that you can now include a "product" in the metadata since fma.owl is again accessible. Option #1 seems like the best near-term solution until we all can figure out the issue of purls. I like the suggestions in #2 but will need to understand the implications better. And you are probably right that we don't have the resources for #3, although I would be interested in being  involved in some way in the process.

I'll keep looking into this, and talk to others on the FMA team, maybe more with you and other members, in order to better understand these options.

**Comment by brinkleyjf** [comment]
Chris etal,
            Sorry I've taken so long to follow up on this, but if I'm honest I think my delay is in itself a response. I have finally gotten around to talking with the main developers of the FMA: (Cornelius Rosse, Onard Mejino, Todd Detwiler, and Melissa Clarkson), and they basically concur with what follows. 
 
There is currently no one working on the "official" UW version of the FMA, and there is not likely to be anyone working on it. We are all retired or working elsewhere, and I (and I believe all of us) have become distracted by too many other things. Thus, the UW can't meet many of your criteria for inclusion in OBO, one of the most important being that the team be responsive to requests for change. I just don't see that happening, nor do I foresee anyone from our group taking on a task like converting the FMA URIs to OBO URIs.  So here are some suggestions from us regarding: 1) The current UW FMA, and 2) FMA-OBO as the vehicle for moving forward with FMA content.
 
The current FMA
This is essentially your option #1, treating the FMA as an inactive artifact that nevertheless could be of use to others because of its history and the availability of FMA Ids.  As long as Bioportal is running the FMA should continue to be available there. And it will also continue to be available at our own site via a browser and SPARQL endpoint as long as the server I've been maintaining stays up. So far, I plan to keep that server going since it requires minimal effort, but I can't guarantee how long that will go on. If possible, you might use the Bioportal address as the main place to get the FMA and add the UW address as a secondary source (with the notation that the versions are the same in both cases).
 
FMA-OBO
Many people, including you,  have pointed out issues with the current FMA that make it difficult to add it as a full-fledged member of OBO. So, one possibility might be for someone in OBO to make a clone of the current FMA, called FMA-OBO, and fix some of the issues that have been raised. These would primarily include changing all the URIs to conform to OBO conventions, fixing errors that prevent the reasoner from working, and submitting FMA-OBO to the OBO GitHub repository.  This is essentially your option 2, except that it would be for the clone FMA-OBO, not the original UW FMA.
 
Ideally, someone could then take over FMA-OBO and respond to requests for change. If someone becomes available to do this, the UW would be willing to freeze the UW FMA to the current version 5.0 (it’s essentially frozen already), and if anyone, including Onard, wants to add to or edit it, he or she would agree to do so in FMA-OBO via the OBO methods. The UW would also put a note on our web page saying that the UW FMA is frozen at version 5.0, is no longer being developed, and any new developments or requests for change should go through OBO and OBO-FMA. 
 
We would also request that within FMA-OBO, proper credit be maintained to the UW FMA as the original source of FMA-OBO, and a computable mechanism be established for relating FMA IDs to FMA-OBO – possibly as annotations within FMA-OBO, a translation service, or both.
 
All of us appreciate your interest in the FMA and your desire assure its viability among the OBO family. We hope that it will be possible to implement the modifications needed to transform the current version of FMA into FMA-OBO.
 
Any thoughts?
 
Jim - and the rest of the FMA team

**Comment by cmungall** [comment]
Thanks @brinkleyjf and the FMA team! This all makes sense.

Here is my recommended action:

- we declare the current record inactive; also add a note that it can't be used for reasoning
- we have the http://purl.org/sig/ont/fma.owl as the official PURL
- we have our own OBO PURLs redirect to this (I think with a 301 code, i.e permanent redirect?)
- we add a product called fmalite that points to the current contents of http://purl.obolibrary.org/obo/fma.owl (i.e. my conversion from 15 years ago, at http://ontologies.berkeleybop.org/fma.owl), that is clearly disambiguated from the official FMA, but is available for groups that need continuity

**Comment by brinkleyjf** [comment]
Chris,
Your recommendations sound fine, but I don't hear much about FMA-OBO, which would be yet a third version of the FMA: 1) FMA-UW,  the original FMA, as per your first three points above; 2) FMA lite, which is the old version you already have, and is very outdated, but could be needed for continuity as your note in your 4th point; and 3) FMA-OBO, which would be a complete copy of the current FMA-UW Version 5.0, with URI's replaced by OBO URIs, and reasoning problems fixed. Sounds like your last point indirectly refers to that. 

In your comment on March 3, your point 2 suggests "the OBO/anatomy team creates a variant of the official FMA that has OBO PURLs, fixes the unsats, and potentially rewires some of the relations to RO. We commit to syncing this with any future releases of FMA (which may be unlikely)". 

Is it possible  for you or members of your team to do that? If you did you wouldn't have to worry about syncing with any future releases of the FMA since the UW would commit to freezing any further development beyond version 5.  

At least initially I think this path would mostly be a programming task as you probably already did with OBO lite. Even if no anatomy group wanted to take it on right away this version of the FMA could at least be available as a more OBO-compliant version of FMA-UW, would be accessible via GitHub and the OBO ecosystem, and would be the basis for further development. That in itself would be a great help to us, since we are trying to find a way to pass the torch to a group who can include it in a larger effort than we’ve been able to do with our small team.

In any case, your ideas for the FMA-UW  and FMA lite seem very doable, and from your last point it sounds like OBO will keep the door open for further development. I just think this further development would be more likely if a computationally-derived FMA-OBO were available. 

I and others on our team would be happy to discuss this further.

Jim

**Comment by dosumis** [comment]
Hi all,  Given how long this has sat around, we have decided to request addition to OLS4 in it's current form.  We will need some way to name it to distinguish from the old OBO translation which is still the only FMA on OLS4.  If there is movement on this soon OLS could take FMA with the new IDs, otherwise we will take as-is.

**Comment by dosumis** [comment]
@jamesamcl

**Comment by brinkleyjf** [comment]
David,
As I wrote in my comments on 7/31/23 I was hoping that someone from OBO might take on the programming task of converting FMA ID's to OBO IDs to create FMA-OBO (ie a translation of the full FMA-UW into OBO format), since  the UW won't have the resources to do so in the forseeable future. Sounds like thats not going to happen (and I didn't really expect it to since I think the world is moving on to other things since the FMA was active), so I agree with your plan to incorporate FMA-UW as is into OLS  - which I presume means Ontology Listing Service - keeping the original UW FMAIDs.

So that would mean there will be two versions of the FMA accessible from OBO (or OLS):
1. FMA-UW, frozen at version 5.0, and currently available at Bioportal and our UW website http://purl.org/sig/ont/fma.owl 
2. FMA-lite, the OBO compliant version that Chris Mungall created several years ago.

If anyone ever does programmatically create a mostly OBO compliant version of FMA-UW (which I called FMA-OBO above), this could replace or augment FMA-UW in OBO. But until and if that happens I think your plan to incorporate FMA-UW as is (without OBO-compliant IDs)  make the most sense.

(And thanks for doing so).

Jim

**Comment by nlharris** [comment]
Can this be closed?

**Comment by cmungall** [comment]
The situation is still exactly the same as in 2015. I made a proposal on
Dec 15 2021, and this is still the course of action I think we should take

**Comment by graybeal** [comment]
After reading through the thread 1.5 times, and as always representing a relatively naive user, a few thoughts. TLDR: I prefer an unambiguous obsoletion of the ontology with respect to the OBO Library, but am open-minded about the best way of solving resolution forwarding issues.

1. Re @cmungall comment last week, does [this comment of Feb 7 2023](https://github.com/OBOFoundry/OBOFoundry.github.io/issues/21#issuecomment-1420982339) update the [recommendation of Dec 14 2021](https://github.com/OBOFoundry/OBOFoundry.github.io/issues/21#issuecomment-994155626)?  At least the follow-on comments resolve the identified issue of original FMA not resolving, but I think that leaves two stated options: _Keep a live entry in OBO with a link to bioportal_, and _mark entry as deprecated_ . Whereas your original 2021 recommendation seemed to be a mix of those two.
2. From the standpoint of "What is the OBO Library collection?", having a non-obsolete, semi-functional FMA entry "in the OBO Library" feels unhelpful; it's a bit of a tripwire for new users to process whether/how they should use the related entities. To me a cleaner solution is that the original FMA is obsolete (because it doesn't follow the requirements), the FMA-lite can be referenced by the obsolete entry but like the original does not qualify for OBO Library status, because not updated.

That leaves two open questions: resolution of some OBO FMA purls, and the creation of a new OBO-compliant version of FMA. The latter should remain an open question, to be resolved if/when someone decides they want to do the necessary work/take on the necessary responsibility for the 'new FMA'.  They can negotiate those hurdles at that time.

I don't know enough about all the existing FMA resolution mechanisms and the use cases they support, but the principles involved seem to be these: (a) being obsolete wouldn't necessarily preclude resolving/forwarding previously existing identifiers  (b) for the ontology itself at least, pointing to BioPortal's entry seems entirely reasonable. And (c) If some resolution/forwarding use cases can't be readily addressed, at times that is in the nature of the technologies we are supporting and their necessary evolution, particularly as ontologies become obsolete. We may not have the luxury of simultaneously implementing ontology obsoletion, with all that should imply for existing use cases, while continuing to fully support those use cases.

There will be losses whichever path is taken (both in time spent, and user impact). My philosophy would be leave things clean, even if that breaks some things, because it isn't too expensive now and it saves time in the future.

**Comment by brinkleyjf** [comment]
Probably ought to add me to the contacts along with Onard since I'm maintaining the Structural Informatics Group (SIG) website where the latest version of the FMA is made available as http://purl.org/sig/ont/fma.owl. As noted above this is version 5.0 of the FMA, and no new versions have been released in the last several years because most members of the group have retired or moved on. The Bioportal version and the SIG version are the same, as noted above by jvendetti, so both have unsatisfiable classes.  Also I recently confirmed to OLS that this is the source version. So I'm pretty sure that Bioportal and OLS both access this version, and it makes sense, as Chris recommended earlier, for  this version to be the official source, with the caveat that it has unsatisfiable classes.

That said, it would be great if someone could generate a version that got rid of the unsatisfiable classes while keeping the original classes as much as possible, and in the process documenting what changes were made. I could then run that by Onard and possibly release a new version. Thats not likely to happen for some time, so in the meantime I think there should only be the one official source. 

As to a history of FMA versions that would be great. There is some information on our SIG website , but it needs to be updated at some point.

Jim Brinkley
brinkley@uw.edu

**Comment by cmungall** [comment]
Thanks everyone for being engaged on this epic ticket!

@jvendetti 
> BioPortal doesn't load the [FMA ontology](https://bioportal.bioontology.org/ontologies/FMA) from the UMLS. Historically, it's been submitted to us as an OWL file by various folks at the Univ. of Washington. No new OWL files have been uploaded in the last 7 years. The current contact is listed as [Onard Mejino](https://github.com/Onard).

Aha, my mistake, thanks! I see indeed that when I download the OWL from bioportal it is the same, and it has the same unsatisfiable classes. Like OLS, BioPortal doesn't run reasoning over submitted OWL and doesn't care about unsatisfiable classes (which is quite sensible but leads to some interesting questions that will only distract us here).

But it looks like the metadata record (which is maintained independently of the header of the OWL file) still says it is provided by the UMLS:

https://bioportal.bioontology.org/ontologies/FMA

<img width="990" height="583" alt="Image" src="https://github.com/user-attachments/assets/a577f987-13fc-4169-9cf4-f05b1da33d62" />

I don't know if you need Onard to make that edit or if you're able to do it directly. I'd recommend just doing it directly, and accurately reflecting the source license since the restrictive UMLS license has nothing to do with this any more.

This changes my recommendations:

Immediate:

* OBO declares http://purl.org/sig/ont/fma.owl as the official source, _but clearly notes this has unsatisfiable classes_ (OBO team)
* This is the SoT for ontology portals and for applications
* OntoBee and AberOWL switch to this version (@leechuck @yongqunh) **provided reasoning is not performed**
* Applications can use the official purl or download from a portal, the results should be the same
 
I think we do this we can finally close this issue. There are various minor discrepancies in ontology metadata that I think are very minor compared to the current situation.

**Comment by brinkleyjf** [comment]
Thanks Jennifer. I tested the editor and was able to add the creative commons license v3 since that's currently what we have on our site. Will update to v4 once we've looked at it. And I'll pass your message on to Onard.

**Comment by trapperkeeper** [comment]
Sorry for being late to the conversation. I'm taking a look at this now.
Since the Structural Informatics Group is no longer developing the FMA, and
everyone (including myself) are busy with other projects, I didn't want to
commit time to fixing the reasoning issues until I was sure I could make
time. Melissa Clarkson and I are currently working at the University of
Kentucky and still use and take inspiration from the FMA. We have decided
to commit the time to making the FMA coherent. The goal ihas been to do so
with as few other changes as possible (for example if an axiom object leads
to a violation of the range of a property, and there is not another obvious
object class, we won't be creating one).

Anyway, this has taken longer than I expected (I've been working on it for
the last couple of days). I think I have a candidate version, as well as a
document of changes that I made. I'm going to run this by the FMA group, to
see if they want to sanction this as a version 5.1.0. I will get back to
this list once that discussion has taken place.

-Todd Detwiler

**Comment by brinkleyjf** [comment]
Todd - this is fantastic! As I mentioned in a separate email Onard and I will look at the new version and, assuming its OK with Onard, I'll work on releasing it as a new version. Might take awhile since I'm in the process of upgrading our website, but I'll get there. Also I'm very interested in Chris's approach using an AI agent, and his suggestions for storing the FMA in GitHub. Could be very useful for future releases and potential collaboration.

**Comment by brinkleyjf** [comment]
Chris et al,
      We (Todd Detwiler and I) have now released FMA 5.1.0, which is satisfiable under EL reasoning, thanks to Todd’s work as detailed in earlier comments. We have also created a  GitHub repository, which now is the primary source for the FMA [https://github.com/uw-sig/FMA](https://github.com/uw-sig/FMA). The current release can be found at  [https://github.com/uw-sig/FMA/releases/tag/5.1.0](https://github.com/uw-sig/FMA/releases/tag/5.1.0).  This release contains fma.owl, puns (pun_fma.owl), the creative commons license, a README, and details about the changes Todd made [https://github.com/uw-sig/FMA/blob/main/docs/Fixing_reasoning_in_the_FMA_revised.pdf](https://github.com/uw-sig/FMA/blob/main/docs/Fixing_reasoning_in_the_FMA_revised.pdf).

More information is at [http://si.washington.edu/info-books/project-details/foundational-model-of-anatomy/summary-of-fma-resources/](http://si.washington.edu/info-books/project-details/foundational-model-of-anatomy/summary-of-fma-resources/).

The purl we have been using , [http://purl.org/sig/ont/fma.owl](http://purl.org/sig/ont/fma.owl), is still the correct IRI for the full FMA (not the limited OBO version). The purl now resolves to the same version of the fma (5.1.0) as in GitHub, but which is  stored  in a GitHub clone on our Structural Informatics Group (sig)  machine, [http://si.washington.edu/](http://si.washington.edu/). We did this rather than changing the purl to point to  GitHub directly because at about 200 mb the FMA is greater than the GitHub limit of 100mb, so is stored using GitHub Large File Storage (LFS),  and that service incurs costs if we exceed their limits - costs we have no budget for. 

We also have not opened up the GitHub repository to pull requests because for now we don’t have anyone to respond to these requests, which is why the FMA should continue to be designated inactive. But at least the creation of the GitHub repository is a step in the right direction if the FMA ever does become active again.

Please let us know if you have any questions, comments or suggestions. Todd and I  have been putting some time into this, so until Todd or I  get onto other things, we might have a little time to respond.

Jim Brinkley and Todd Detwiler

**Comment by cmungall** [comment]
My proposal:

* the OBO record is marked inactive
* we redirect fma.owl here https://github.com/OBOFoundry/purl.obolibrary.org/blob/master/config/fma.yml to  http://purl.org/sig/ont/fma.owl
* we redirect OBO class purls to the official FMA purls
* we record the history in the text

The advantage is that FMA doesn't blink out of existence in OLS and other portals that use the OBO registry as intended. 

It's obviously a deviation from the OBO guidelines on PURLs, but FMA predates OBO.

**Comment by matentzn** [comment]
Slide deviation:

Phase 1 (self contained):

- the OBO record is marked **obsolete** to clearly indicate to not use it (it already is "inactive") 
- ✅ we redirect fma.owl here [OBOFoundry/purl.obolibrary.org@master/config/fma.yml](https://github.com/OBOFoundry/purl.obolibrary.org/blob/master/config/fma.yml?rgh-link-date=2026-08-20T19%3A25%3A56.000Z) to http://purl.org/sig/ont/fma.owl
- ✅ we record the history in the text

Phase 2 (strongly suggested optional, but not a dependency the above):

- we redirect OBO class purls to the official FMA purls (assuming that is at all possible using a URL rewrite - this requires FMA ids to be identical to FMA OBO IDs, which I hope that is the case).

@jamesaoverton Can the second (redirect OBO class purls to the official FMA purls) be done with the current setup technically? If so, I will make a specific issue for this and go ahead an implement it.

**Comment by matentzn** [comment]
As soon as I get @cmungall @jamesaoverton confirms I will get going on these

**Comment by jamesaoverton** [comment]
I'm ok with this plan.

> Can the second (redirect OBO class purls to the official FMA purls) be done with the current setup technically?

Yes, it should work the same as a custom term browser override, examples here: https://github.com/OBOFoundry/purl.obolibrary.org/blob/master/config/obo.yml#L49

**Comment by cmungall** [comment]
If we mark the OBO record as obsolete then this would be interpreted as the FMA being obsolete, which is manifestly not the case. The true state of FMA is inactive.

Furthermore, marking it as obsolete would remove it from browsers like OLS, which would be a disservice to the FMA and to the community. This is easily rectified by having it separately registered but someone needs to coordinate this

**Comment by brinkleyjf** [comment]
I like Chris's plan, including having the OBO URI's redirect to http://purl.org/sig/ont/fma.owl since that is the official site of the latest FMA. I also like the idea that the FMA is designated as inactive, not obsolete since I believe many sites still reference FMA IDs. Not sure how you all could designate the OBO version as obsolete but the Sig version as inactive. One thing that might be of interest is this thread from OLS: https://github.com/EBISPOT/ols4/issues/1101. At that time (2025) the FMA was registered there, but the URI was not resolving, so I suggested that they replace the IRI with http://purl.org/sig/ont/fma.owl. I didn't follow up so I don't know if they actually did that. But just now I searched for FMA in OLS and its not listed at all. So do we need to re-register it from our side like Chris seems to be suggesting?

**Comment by matentzn** [comment]
The plan we had was this:

1. Clearly rename the _OBO record_ as "OBO Subset of FMA" and obsolete it
2. Mention that the main version is not obsolete and point where to find it
3. Redirect OBO FMA purls to the non-obsolete version to keep some level of stability for adopters of the OBO style IDs.

Note:

1. FMA can be added to OLS independently of OBO Foundry and it will stay in BioPortal anyways
2. The metadata record still points clearly to the active version so people can find it

So our disagreement here is this:

Is it more or less confusing to keep the metadata record in OBO active? The OBO variant of the FMA is not the same as the original. Its a different file. Its like the NCIT obo subset, which we could decide to obsolete independently of NCIT.

I think its cleaner to obsolete but I also dont care so much. One more comment to making it inactive and I will do that :)

@brinkleyjf I just checked and it _seems_ to me it is correctly loaded from bitbucket: https://www.ebi.ac.uk/ols4/ontologies/fma. Can you confirm?

**Comment by brinkleyjf** [comment]
Yes you're right. I missed it when I searched before. And the FMA listing does resolve to the correct URI,  http://purl.org/sig/ont/fma.owl. That purl goes to our Structural Informatics Group (sig) server hosted at the University of Washington. I know Todd Detwiler once hosted the FMA in Bitbucket, but I don't see that the current OLS entry goes there. BTW individual FMA IDs, like http://purl.org/sig/ont/fma/fma61775 go to the corresponding entry on Todd's FMA browser, also hosted on our site. (At least while I'm still maintaining it in my retirement).

**Comment by matentzn** [comment]
@brinkleyjf thank you!

**Comment by leechuck** [comment]
Apologies for the silence on this one. AberOWL still serves the OBO-registry copy (5.0.0, "(subset)"). We are switching to the official FMA 5.1.0 from http://purl.org/sig/ont/fma.owl; tracked in https://github.com/bio-ontology-research-group/aberowl2/issues/92 and we will address it soon. Thanks Jim and Todd for the 5.1.0 release, and Chris and Nico for driving this.

**Comment by matentzn** [comment]
Thank you @leechuck 

@brinkleyjf @Onard, I still maintain that https://github.com/OBOFoundry/OBOFoundry.github.io/pull/2849/changes is the _least_ confusing way to present FMA; Please review and tell me your opinion.

If you prefer, I can set the project to "inactive" - just to get it merged - but I believe deprecating the OBO-specific variant of FMA should be recorded as obsolete.

**Comment by cmungall** [comment]
why not create a new entry fma_subset, born obsolete. But IMO even this is
overkill.

the FMA itself is not obsolete and OBO shouldn't say this

**Comment by matentzn** [comment]
> the FMA itself is not obsolete and OBO shouldn't say this

We were not, but since this discussion went on for too long I now enacted your suggested changes. I just want to state that I personally weakly disagree with the outcome of this discussion due to the unclear nature of the FMAs role in OBO (no OBO PURLs for IDs, no OBO style licensing, no relationship to other OBO ontologies AND, at the same time, a hugely important domain now co-habitated with Uberon).

Remaining steps: 

1. Approve https://github.com/OBOFoundry/OBOFoundry.github.io/pull/2849
2. Approve https://github.com/OBOFoundry/purl.obolibrary.org/pull/1125
